test/tla/ChangRobertsTyped_Test.tla

```diff
@@ -0,0 +1,103 @@
+--------------------- MODULE ChangRobertsTyped_Test -------------------------
+(*
+ * A test setup for ChangRobertsTyped.
+ *)
+EXTENDS Integers, Apalache
+
+\* a copy of constants from ChangRobertsTyped
+CONSTANTS
+    \* @type: Int;
+    N,
+    \* @type: Int -> Int;
+    Id
+
+\* a copy of state variables from ChangRobertsTyped
+VARIABLES
+    \* @type: Int -> Set(Int);
+    msgs,
+    \* @type: Int -> Str;
+    pc,
+    \* @type: Int -> Bool;
+    initiator,
+    \* @type: Int -> Str;
+    state
+
+INSTANCE ChangRobertsTyped
+
+\* We bound N in the test
+MAX_N == 6
+
+\* we override Node, as N is not known in advance
+OVERRIDE_Node == { i \in 1..MAX_N: i <= N }
+
+\* initialize constants
+ConstInit ==
+    /\ N \in 2..MAX_N
+    /\ Id \in [ 1..MAX_N -> Int ]
+
+\* The below constraints are copied from ASSUME.
+\* They are not enforced automatically, see issue #69.
+Assumptions ==    
+    /\ Node = DOMAIN Id
+    /\ \A n \in Node: Id[n] >= 0
+    /\ \A m,n \in Node : m # n => Id[m] # Id[n]  \* IDs are unique
+
+InitAndAssumptions ==
+    Init /\ Assumptions
+
+(********************** STATELESS OPERATOR TESTS ******************************)
+
+\* Note that succ(n) is not referring to state variables,
+\* so we can test it in isolation.
+\*
+\* @testStateless
```

Comment:
Could we go with `@test` instead of `@testStateless`? Or do we predict that `@test` could be used in the future for something else?

Reply:
I don't know. I think this is actually a rare case, when we have a stateless operator. When I was writing this text, it was not even clear to me, why cannot we directly tag `succ` with `@require` and `@ensure`.

Reply:
In the MBT for IBC, we have a single module, [IBC](https://github.com/informalsystems/ibc-rs/blob/master/modules/tests/support/model_based/IBC.tla), changing the state.
The remaining modules ([ICS02](https://github.com/informalsystems/ibc-rs/blob/master/modules/tests/support/model_based/ICS02.tla) and [ICS03](https://github.com/informalsystems/ibc-rs/blob/master/modules/tests/support/model_based/ICS03.tla)) only contain stateless operators that are then invoked by the IBC one.

I don't know if this is a common idiom, however.

Reply:
It's reasonable to expect that people would write larger specs like this. So far, I have not seen it often. It may also be a difference between people who come from imperative and functional languages.